rotkehlchen/utils/datadir.py

```diff
@@ -0,0 +1,136 @@
+    ASSETIMAGESDIR_NAME,
+    AVATARIMAGESDIR_NAME,
+    CUSTOMASSETIMAGESDIR_NAME,
+    GLOBALDIR_NAME,
+    IMAGESDIR_NAME,
+    MISCDIR_NAME,
+    USERDB_NAME,
+    USERSDIR_NAME,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.utils.misc import ts_now
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _create_directory_with_potential_backup(data_dir: Path, name: str) -> Path:
+    """Create a subdirectory under the main data directory. If the directory
+    already exists and has a user database in it, that means that someone made
+    a user with that name. In which edge case we will try to rename and backup"""
+    subdir = data_dir / name
+    if (subdir / USERDB_NAME).exists():
+        # someone had a user, called name ... and did not get moved with initial for loop
+        try:  # this should not happen actually ...
+            copytree(subdir, f'{ts_now()}_backup_user')
+        except (PermissionError, Error) as e:
+            log.error(f'During restructuring data directory failed to make a backup of a user called {name} due to {e}')  # noqa: E501
+        else:
+            rmtree(subdir, ignore_errors=True)
+
+    subdir.mkdir(parents=True, exist_ok=True)
+    return subdir
+
+
+def _move_paths_to_directory(paths: list[Path], target_dir: Path) -> None:
+    """Move given paths to the target directory"""
+    for path in paths:
+        target_path_dir = target_dir / path.name
+        try:
+            move(path, target_path_dir)
+        except (PermissionError, Error) as e:
+            log.error(f'During restructuring data directory failed to copy {path} to {target_path_dir} due to {e}')  # noqa: E501
+
+
+def _handle_images(data_dir: Path, icons_path: Path | None) -> None:
+    if icons_path is not None:
+        # create the images data directory and move paths there
+        img_data_dir = _create_directory_with_potential_backup(data_dir, IMAGESDIR_NAME)
+        assets_dir = img_data_dir / ASSETIMAGESDIR_NAME
+        assets_dir.mkdir(parents=True, exist_ok=True)
+
+        targetpath = assets_dir / ALLASSETIMAGESDIR_NAME
+        try:
+            move(icons_path, targetpath)
+        except (PermissionError, Error) as e:
+            log.error(f'During restructuring data directory failed to move {icons_path} to {targetpath} due to {e}')  # noqa: E501
+            return
+
+        all_assets_dir = targetpath
+        for name, target_dir in ((AVATARIMAGESDIR_NAME, img_data_dir), (CUSTOMASSETIMAGESDIR_NAME, assets_dir)):  # noqa: E501
+            subdir = all_assets_dir / name
+            if subdir.exists():
+                targetpath = target_dir / name
+                try:
+                    move(subdir, targetpath)
+                except (PermissionError, Error) as e:
+                    log.error(f'During restructuring data directory failed to move {subdir} to {targetpath} due to {e}')  # noqa: E501
+                    continue
```

Comment:
don't know if you prefer to reduce the indentation modifying the flow but

```suggestion
            if not subdir.exists():
                continue
            targetpath = target_dir / name
            try:
                move(subdir, targetpath)
            except (PermissionError, Error) as e:
               log.error(f'During restructuring data directory failed to move {subdir} to {targetpath} due to {e}')  # noqa: E501
```

also the last continue isn't needed since there isn't anything after ir

Reply:
no the old code is better here. Readability above all.

Reply:
hm no no no no. I thought it's like the other suggestion in my previous comment. I am applying this change too.